.github/scripts/icomoon_peek.py

```diff
@@ -1,33 +1,63 @@
+from typing import List
+import re
 from selenium.common.exceptions import TimeoutException
 
 # pycharm complains that build_assets is an unresolved ref
 # don't worry about it, the script still runs
 from build_assets.SeleniumRunner import SeleniumRunner
-from build_assets import filehandler, util
+from build_assets import filehandler, arg_getters
 
 
 def main():
-    args = util.get_commandline_args()
+    args = arg_getters.get_selenium_runner_args(True)
     new_icons = filehandler.find_new_icons(args.devicon_json_path, args.icomoon_json_path)
+
+    # get only the icon object that has the name matching the pr title
+    filtered_icons = find_object_added_in_this_pr(new_icons, args.pr_title)
+
+    # print list of new icons
+    print("List of new icons:", *new_icons, sep = "\n")
+    print("Icons being uploaded:", *filtered_icons, sep = "\n")
+
     if len(new_icons) == 0:
         print("No files need to be uploaded. Ending script...")
         return
 
-    # print list of new icons
-    print("List of new icons:", *new_icons, sep = "\n")
+    screenshot_folder = filehandler.create_screenshot_folder("./") 
+    if len(filtered_icons) == 0:
+        print("No icons found matching the icon name in the PR's title. Fallback to uploading all new icons found.")
+        screenshot_folder = ""
 
     runner = None
     try:
         runner = SeleniumRunner(args.download_path, args.geckodriver_path, args.headless)
-        svgs = filehandler.get_svgs_paths(new_icons, args.icons_folder_path)
-        runner.upload_svgs(svgs)
+        svgs = filehandler.get_svgs_paths(filtered_icons, args.icons_folder_path)
+        runner.upload_svgs(svgs, screenshot_folder)
         print("Task completed.")
     except TimeoutException as e:
+        print("Selenium Time Out Error: ", e.stacktrace, sep='\n')
+    except Exception as e:
         print(e)
-        print(e.stacktrace)
     finally:
         runner.close()
 
 
+def find_object_added_in_this_pr(icons: List[dict], pr_title: str):
+    """
+    Find the icon name from the PR title. 
+    :param icons, a list of the font objects found in the devicon.json.
+    :pr_title, the title of the PR that this workflow was called on.
+    :return a list containing the dictionary with the "name"
+    entry's value matching the name in the pr_title.
+    If none can be found, return an empty list.
+    """
+    try:
+        pattern = re.compile(r"(?<=^new icon: )\w+ (?=\(.+\))", re.I)
+        icon_name = pattern.findall(pr_title)[0].lower().strip()  # should only have one match
```

Comment:
I'm not sure if this method of finding the icon in the title is very stable.. We should give some hints (in the comment of the Bot?) to ensure the title of a pull request is right or maybe we can think of a better solution?

Reply:
I agree with you regarding using the title. I couldn't think of any better way to find the icon that's being included in this commit.

Maybe you can help me think of a better way to do this?

What I've tried:
- Using git diff to find out which folders were added in this PR => upload only those
  - Many PR have multiple commits and git diff needs two different commit SHAs (the current one and the one before the PR) to compare. I couldn't figure out a way to find the commit SHA before the PR.

I'm looking into:
- Using the GitHub Action pull_request context. However, the docs is a bit complicated so I'm not too sure what I can find that's useful.

Why I want to do this:
- Make the peek process faster. The longest step in the workflow is the selenium uploading to Icomoon step. By limiting the amount of files being uploaded, it makes reviewing the PR faster for everyone (usually by about 30 s which I admit is not a lot)
- Let us know which icon to take up close screenshots on. I don't think we should take screenshots for every single icons every time we run the peek workflow since it'd slow down the workflow.

Reply:
Currently I don't have a better solution ready (though about `git diff` as well).. I think it's okey to stick with the title for now. Maybe letting the Bot add a comment which points this out if it could not be found?

Reply:
Yeah, that sounds like a good idea. I'll add it in for now and see if we can find a better way. There's something promising in the docs.